om/json.go

```diff
@@ -145,6 +145,31 @@ func (r *JSONRepository[T]) CreateIndex(ctx context.Context, cmdFn func(schema F
 	return r.client.Do(ctx, cmdFn(r.client.B().FtCreate().Index(r.idx).OnJson().Prefix(1).Prefix(r.prefix+":").Schema())).Error()
 }
 
+// CreateAndAliasIndex creates a new index, aliases it, and drops the old index if needed.
+func (r *JSONRepository[T]) CreateAndAliasIndex(ctx context.Context, cmdFn func(schema FtCreateSchema) rueidis.Completed) error {
+	// Create a new index
+	newIndex := r.idx + "_new"
+	alias := r.idx + "_alias"
+	if err := r.client.Do(ctx, cmdFn(r.client.B().FtCreate().Index(newIndex).OnJson().Prefix(1).Prefix(r.prefix+":").Schema())).Error(); err != nil {
+		return err
+	}
+
+	// Alias the new index
+	if err := r.client.Do(ctx, r.client.B().FtAliasadd().Alias(alias).Index(newIndex).Build()).Error(); err != nil {
+		return err
+	}
+
+	// Drop the old index
+	if err := r.DropIndex(ctx); err != nil {
+		return err
+	}
+
+	// Update the repository index name
+	r.idx = newIndex
```

Comment:
It seems like a user is required to always invoke `CreateAndAliasIndex` at startup for `Search` to work correctly. Furthermore, a user can't invoke `CreateAndAliasIndex` more than once during the lifetime of a repository; otherwise, the index and alias names will be appended the `_new` and `_alias` suffix multiple times.

Can we allow the following scenario?
1. `CreateAndAliasIndex` is not required at every startup.
2. Allow multiple `CreateAndAliasIndex` invocations during the same lifetime of a repository.

Reply:
Correct, take a look at it once, I am checking it using FT LIST

Reply:
@rueian